js_modules/dagster-ui/packages/ui-core/src/search/useIndexedDBCachedQuery.tsx

```diff
@@ -84,9 +84,16 @@ export function useIndexedDBCachedQuery<TQuery, TVariables extends OperationVari
         expiry: new Date('3000'), // never expire,
       },
     );
-    setData(data);
-    fetchState[key]?.onFetched.forEach((cb) => cb(queryResult));
     delete fetchState[key];
+    const onFetched = fetchState[key]?.onFetched;
+    try {
+      setData(data);
+    } catch (e) {}
+    onFetched?.forEach((cb) => {
+      try {
+        cb(queryResult);
+      } catch (e) {}
```

Comment:
Just calling out that this mutes a potentially significant amount of code since lots of stuff could run downstream of the data arriving - this should probably still emit a datadog trace + console.error just in case!

Reply:
Ah yeah I'll rethrow the error in a setTimeout. We need a general utility for this.